### PR TITLE
List and Create Buckets

### PR DESCRIPTION
**Pull request by sunbryely-influxdata**
List Buckets
Create Buckets

### CODE REVIEW

File: src/common/paths/buckets.yml [hunk 1]
```diff
@@ -3,49 +3,120 @@ get:
...
     - $ref: "../parameters/TraceSpan.yml"
     - $ref: "../parameters/Offset.yml"
     - $ref: "../parameters/Limit.yml"
     - $ref: "../parameters/After.yml"
     - in: query
       name: org
-      description: The name of the organization.
+      description: |
+        Organization name.
+        The name of the organization.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:45]: Depending on which applies, follow this: https://docs.influxdata.com/influxdb/cloud/api/#operation/PostWrite
or this: https://docs.influxdata.com/influxdb/cloud/api/#operation/GetTasks

I'm not sure `org` and `orgID` are used by Cloud at all here b/c they use the token organization to scope the results for most endpoints. We still need to test Cloud and OSS for this case-by-case.

File: src/common/paths/buckets.yml [hunk 2]
```diff
@@ -3,49 +3,120 @@ get:
...
       schema:
         type: string
     - in: query
       name: orgID
-      description: The organization ID.
+      description: |
+        Organization ID.
+        The organization ID.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:52]: Depending on which applies, follow this: https://docs.influxdata.com/influxdb/cloud/api/#operation/PostWrite
or this: https://docs.influxdata.com/influxdb/cloud/api/#operation/GetTasks

I'm not sure `org` and `orgID` are used by Cloud at all here b/c they use the token organization to scope the results for most endpoints. We still need to test Cloud and OSS for this case-by-case.

File: src/common/paths/buckets.yml [hunk 3]
```diff
@@ -3,49 +3,120 @@ get:
...
       schema:
         type: string
     - in: query
       name: name
-      description: Only returns buckets with a specific name.
+      description: |
+        Bucket name.
+        Only returns buckets with a specific name.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:59]:
```suggestion
        Only returns buckets with this name.
```

File: src/common/paths/buckets.yml [hunk 4]
```diff
@@ -3,49 +3,120 @@ get:
...
       schema:
         type: string
     - in: query
       name: id
-      description: Only returns buckets with a specific ID.
+      description: |
+        Bucket ID.
+        Only returns buckets with a specific ID.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:66]:
```suggestion
        Only returns the bucket with this ID.
```

File: src/common/paths/buckets.yml [hunk 5]
```diff
@@ -3,49 +3,120 @@ get:
...
       schema:
         type: string
   responses:
     "200":
-      description: A list of buckets
+      description: |
+        Success.
+        The response body contains a list of buckets and URI links to adjacent
+        buckets.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:74]:
```suggestion
        The response body contains a list of buckets and URIs for additional paged results.
```
?

File: src/common/paths/buckets.yml [hunk 6]
```diff
@@ -3,49 +3,120 @@ get:
...
       description: Unexpected error
       content:
         application/json:
           schema:
             $ref: "../schemas/Error.yml"
+  x-codeSamples:
+    - lang: Shell
+      label: cURL
+      source: |
+        curl --request GET "https://cloud2.influxdata.com/api/v2/buckets?name=_monitoring" \
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:95]:
```suggestion
        curl --request GET "http://localhost:8086/api/v2/buckets?name=_monitoring" \
```

File: src/common/paths/buckets.yml [hunk 7]
```diff
@@ -3,49 +3,120 @@ get:
...
+    Creates a [bucket]({{% INFLUXDB_DOCS_URL %}}/cloud/reference/glossary/#bucket)
+    and returns the created bucket along with metadata. The default data
+    [retention period]({{% INFLUXDB_DOCS_URL %/cloud/reference/glossary/#retention-period}})
+    is 30 days.
+
+    #### InfluxDB OSS
+
+    - A single InfluxDB OSS instance supports approximately 20 buckets
+    actively being written to or queried across all organizations depending on
+    the use case. Any more than that can adversely affect performance.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:114]:
```suggestion
    - A single InfluxDB OSS instance supports
    active writes or queries for approximately 20 buckets across all organizations at a given time. Reading or writing to more than 20 buckets at a time can adversely affect performance.
```
@kelseiv could say this better.

File: src/common/paths/buckets.yml [hunk 8]
```diff
@@ -57,20 +128,78 @@ post:
           $ref: "../schemas/PostBucketRequest.yml"
   responses:
     "201":
-      description: Bucket created
+      description: |
+        Success.
+        Bucket created.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:133]:
```suggestion
        The bucket was created.
```

File: src/common/paths/buckets.yml [hunk 9]
```diff
@@ -57,20 +128,78 @@ post:
...
+      content:
+        application/json:
+          schema:
+            $ref: "../schemas/Error.yml"
+    "401":
+      $ref: "../responses/AuthorizationError.yml"
+    "403":
+      description: |
+        Forbidden.
+        Bucket quota exceeded.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:152]:
```suggestion
        The bucket quota is exceeded.
```
Maybe provide more information about "bucket quota".

File: src/common/responses/example-responses/GetBucketsSuccessResponse.yml [hunk 10]
```diff
@@ -0,0 +1,33 @@
+  successResponse:
+    value: {
+      "links": {
+        "self": "/api/v2/buckets?descending=false\u0026limit=20\u0026name=_monitoring\u0026offset=0\u0026orgID=ORG_ID"
+      },
+      "buckets": [
+        {
+          "id": "77ca9dace40a9bfc",
+          "orgID": "ORG_ID",
```
Comment (jstirnaman) [inline src/common/responses/example-responses/GetBucketsSuccessResponse.yml:9]:
```suggestion
          "orgID": "INFLUX_ORG_ID",
```

File: src/common/responses/example-responses/GetBucketsSuccessResponse.yml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+              "type": "expire",
+              "everySeconds": 604800
+            }
+          ],
+          "createdAt": "2022-03-15T17:22:33.72617939Z",
+          "updatedAt": "2022-03-15T17:22:33.726179487Z",
+          "links": {
+            "labels": "/api/v2/buckets/77ca9dace40a9bfc/labels",
+            "members": "/api/v2/buckets/77ca9dace40a9bfc/members",
+            "org": "/api/v2/orgs/ORG_ID",
```
Comment (jstirnaman) [inline src/common/responses/example-responses/GetBucketsSuccessResponse.yml:25]:
```suggestion
            "org": "/api/v2/orgs/INFLUX_ORG_ID",
```

File: src/common/schemas/PostBucketRequest.yml [hunk 12]
```diff
@@ -1,15 +1,42 @@
   properties:
     orgID:
+      description: |
+        Organization ID.
+        The ID of the organization.
```
Comment (jstirnaman) [inline src/common/schemas/PostBucketRequest.yml:5]: See earlier comment about org params.

File: src/common/schemas/PostBucketRequest.yml [hunk 13]
```diff
@@ -1,15 +1,42 @@
...
       type: string
     name:
+      description: |
+        Bucket name.
```
Comment (jstirnaman) [inline src/common/schemas/PostBucketRequest.yml:9]:
```suggestion
```

File: src/common/schemas/PostBucketRequest.yml [hunk 14]
```diff
@@ -1,15 +1,42 @@
...
+        The name of the bucket.
       type: string
     description:
+      description: |
+        The description of the bucket.
```
Comment (jstirnaman) [inline src/common/schemas/PostBucketRequest.yml:14]:
```suggestion
        A description of the bucket.
```

File: src/common/schemas/PostBucketRequest.yml [hunk 15]
```diff
@@ -1,15 +1,42 @@
...
       type: string
     rp:
+      description: |
+        Retention policy is an InfluxDB 1.x concept that represents the duration
+        of time that each data point in the retention policy persists. Use `rp`
+        for compatibility with InfluxDB 1.x.
+        The InfluxDB 2.x and Cloud equivalent is
+        [retention period]({{% INFLUXDB_DOCS_URL %}}/cloud/reference/glossary/#retention-period).
```
Comment (jstirnaman) [inline src/common/schemas/PostBucketRequest.yml:22]:
```suggestion
        [retention period]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#retention-period).
```

File: src/common/schemas/PostBucketRequest.yml [hunk 16]
```diff
@@ -1,15 +1,42 @@
...
       $ref: "./RetentionRules.yml"
     schemaType:
+      description: |
+        Schema Type.
+        Use `explicit` to enforce column names, tags, fields, and data types for
+        your data.
+
+        #### InfluxDB Cloud
+
+        - Default value is `implicit`.
```
Comment (jstirnaman) [inline src/common/schemas/PostBucketRequest.yml:35]:
```suggestion
        - Default is `implicit`.
```

File: src/common/schemas/PostBucketRequest.yml [hunk 17]
```diff
@@ -1,15 +1,42 @@
...
+
+        #### InfluxDB OSS
+
+        - Does not support `schemaType`.
```
Comment (jstirnaman) [inline src/common/schemas/PostBucketRequest.yml:39]:
```suggestion
        - Doesn't support `schemaType`.
```

File: src/common/schemas/RetentionRule.yml [hunk 18]
```diff
@@ -8,11 +8,27 @@
...
       minimum: 0
     shardGroupDurationSeconds:
       type: integer
       format: int64
-      description: Shard duration measured in seconds.
-  required: [type, everySeconds]
+      description: |
+        The shard group duration.
+        The duration of time or interval that each shard group covers measured
+        in seconds.
```
Comment (jstirnaman) [inline src/common/schemas/RetentionRule.yml:24]:
```suggestion
        The duration or interval (in seconds) that each shard group covers.
```

File: src/oss/tags.yml [hunk 19]
```diff
@@ -11,6 +11,18 @@ tags:
         - [Authorize API requests]({{% INFLUXDB_DOCS_URL %}}/api-guide/api_intro/#authentication).
         - [Manage API tokens]({{% INFLUXDB_DOCS_URL %}}/security/tokens/).
         - [Assign a token to a specific user]({{% INFLUXDB_DOCS_URL %}}/security/tokens/create-token/).
+  - name: Buckets
+    description: |
+      Store your data in InfluxDB [buckets]({{% INFLUXDB_DOCS_URL %}}/cloud/reference/glossary/#bucket).
```
Comment (jstirnaman) [inline src/oss/tags.yml:16]:
```suggestion
      Store your data in InfluxDB [buckets]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#bucket).
```

File: src/oss/tags.yml [hunk 20]
```diff
@@ -11,6 +11,18 @@ tags:
...
+      A bucket is a named location where time series data is stored. All buckets
+      have a [retention period]({{% INFLUXDB_DOCS_URL %/cloud/reference/glossary/#retention-period}}),
```
Comment (jstirnaman) [inline src/oss/tags.yml:18]:
```suggestion
      have a [retention period]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#retention-period),
```

File: src/oss/tags.yml [hunk 21]
```diff
@@ -11,6 +11,18 @@ tags:
...
+      a duration of time that each data point persists. InfluxDB drops all
+      points with timestamps older than the bucket’s retention period.
+      A bucket belongs to an organization.
+
+      ### Related guides
+
+        - [Manage buckets]({{% INFLUXDB_DOCS_URL %}}/cloud/organizations/buckets/)
```
Comment (jstirnaman) [inline src/oss/tags.yml:25]:
```suggestion
        - [Manage buckets]({{% INFLUXDB_DOCS_URL %}}/organizations/buckets/)
```

**Review by jstirnaman — CHANGES_REQUESTED** [review]

File: src/common/paths/buckets.yml [hunk 22]
```diff
@@ -3,49 +3,120 @@ get:
...
+          --header "Authorization: Token INFLUX_TOKEN" \
+          --header "Accept: application/json" \
+          --header "Content-Type: application/json"
 post:
   operationId: PostBuckets
   tags:
     - Buckets
   summary: Create a bucket
+  description: |
+    Creates a [bucket]({{% INFLUXDB_DOCS_URL %}}/cloud/reference/glossary/#bucket)
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:105]:
```suggestion
    Creates a [bucket]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#bucket)
```

File: src/common/paths/buckets.yml [hunk 23]
```diff
@@ -3,49 +3,120 @@ get:
...
+    and returns the created bucket along with metadata. The default data
+    [retention period]({{% INFLUXDB_DOCS_URL %/cloud/reference/glossary/#retention-period}})
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:107]:
```suggestion
    [retention period]({{% INFLUXDB_DOCS_URL %}}/reference/glossary/#retention-period)
```

**Review by jstirnaman — CHANGES_REQUESTED** [review]

File: src/common/paths/buckets.yml (hunk 9 above)
Comment (sunbryely-influxdata) [reply src/common/paths/buckets.yml:152]: Will provide bucket quota information in the Limitations section

File: src/common/paths/buckets.yml (hunk 5 above)
Comment (sunbryely-influxdata) [reply src/common/paths/buckets.yml]: Changed this to `The response body contains a list of buckets.` and provided additional description regarding the links in `Links.yml`

File: src/common/paths/buckets.yml [hunk 24]
```diff
@@ -2,50 +2,148 @@ get:
...
+
+        #### InfluxDB Cloud
+
+         - Doesn't use `org` or `orgID`.
+         - Creates a bucket in the organization associated with the authorization (API token).
+
+         #### InfluxDB OSS
+
+         - Accepts either `org` or `orgID`.
+         - InfluxDB creates the bucket within this organization.
```
Comment (jstirnaman) [inline src/common/paths/buckets.yml:55]: 👍

**Review by jstirnaman — APPROVED** [review]
LGTM

**Pull request merged**